### PR TITLE
feat(SPDX): Use new SPDX library (#1496)

### PR DESCRIPTION
**Pull request by tuannn2**
Signed-off-by: Nguyen Nhu Tuan <tuan2.nguyennhu@toshiba.co.jp>

[//]: # (Copyright Bosch.IO GmbH 2020)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

> Please provide a summary of your changes here.
> * Which issue is this pull request belonging to and how is it solving it? (*Refer to issue here*)
> * Did you add or update any new dependencies that are required for your change?

Issue: #1496

#### Import SPDX BOM in Project page
Use file [bom.spdx.rdf.zip](https://github.com/eclipse/sw360/files/9158059/bom.spdx.rdf.zip)

Steps
1. Go to project page
2. Click "Import SPDX BOM" button
3. Upload bom.spdx.rdf file

Validate
- "asoftware (1.1)" project was created
- "Browser", "Buffer" and "CompressionEng" components were created
- "Browser(2.1)", "Buffer(2.2)" and "CompressionEng 3.1" were created
- linked releases of "asoftware (1.1)" project are Browser(2.1) and Buffer(2.2)
- linked release of "Browser(2.1)" is "CompressionEng 3.1"

#### Import SPDX BOM in Component page
Use file [bom.spdx.rdf.zip](https://github.com/eclipse/sw360/files/9158062/bom.spdx.rdf.zip)

Steps
1. Go to component page
2. Click "Import SPDX BOM" button
3. Upload bom.spdx.rdf file

Validate
- "asoftware", "Browser", "Buffer" and "CompressionEng" components were created
- "asoftware (1.1)", "Browser(2.1)", "Buffer(2.2)" and "CompressionEng 3.1" were created
- linked releases of "asoftware (1.1)" release are Browser(2.1) and Buffer(2.2)
- linked release of "Browser(2.1)" is "CompressionEng 3.1"

#### Import SPDX license to release
Use file 
[SPDX2_zlib128.zip_example-spdx.2.rdf.zip](https://github.com/eclipse/sw360/files/9158080/SPDX2_zlib128.zip_example-spdx.2.rdf.zip)

Steps
1. Go to the edit page of Release Buffer(2.2)
2. In Attachments tab, add attachment SPDX2_time-1.9.tar.gz_1535120734-spdx.rdf
3. Click "Update Release" button
4. In the view page of Release Buffer(2.2), click Fossology icon to scan license on "Clearing details" tab
5. Click "Show License Info" button
6. Click "Add data to this release" button

Validate
- At step 4th, the result same as the image

afterScanFossology

- At step 6th, the result same as the image
AfterAddLicensetoRelease

#### Generate License Info in Project page

Steps
1. Go to the details page of "asoftware (1.1)" project
2. In "License Clearing" tab, click Generate License Info (project only) link

Validate
- Readme generation was created same as the image
Readme_genaration

### Checklist
Must:
- [ ] All related issues are referenced in commit messages and in PR

### CODE REVIEW

**Comment by KoukiHama** [comment]
This PR does not change any SW360 functionality, it just changes the SPDX library used by SW360.
Testing should confirm that all existing SPDX related functionality is working fine.

**Comment by tienlee** [comment]
I would like to attach the slides which are used in our telco today.
Let's keep to discuss the json format between anchore/syft and a perfect SPDX file (SPDXJSONExample-v2.2.spdx.json)

[SW360-20220831 - Demo1594.pptx](https://github.com/eclipse/sw360/files/9460025/SW360-20220831.-.Demo1594.pptx)

**Comment by akapti** [comment]
Hi @tuannn2 
I feel that rebase didn't went well, because PR contains changes which is totally unrelated to SPDX library upgrade.

for example:

It includes changes related to ChangeLogs in following file: https://github.com/eclipse/sw360/pull/1594/files#diff-29f4181fef7f3e6f651dde38f70234ffa90ccc0f7cfebb9cdd258071f8aa2e01 , https://github.com/eclipse/sw360/pull/1594/files#diff-e06edfe0b28e68467281adcf719a12122b6ca9565682e3523ea33ed4d39a52c7, https://github.com/eclipse/sw360/pull/1594/files#diff-7b84421f5372707112717574bd301252e3c831d747c1164593d83a2dbde26acf and few more files.


also the PR contains changes related to Thrift configuration in following files: https://github.com/eclipse/sw360/pull/1594/files#diff-c70fecc2e6dc5d8dc0441ca5fa4c26f109ebd954b2192cf3b4b5def24e24f23f

also removed changes related to Reading Obligation from README OSS in following files: https://github.com/eclipse/sw360/pull/1594/files#diff-74ca1a57d71f7fae13d891c82005a0a933bc9d39ab88b105ac531dc50f31712eL2706 and https://github.com/eclipse/sw360/pull/1594/files#diff-a47079862cba8e2c7eb02695e28dcd5e1016bdca7338e88075985b5dcb0e80baL308

Let me know in case I am wrong or if I missed something.
Or Can you please explain why changes related to ChangeLogs, Thrift configuration and ReadMe OSS obligations are part of this PR?

**Comment by tuannn2** [comment]
Thank you for your reviewing, @akapti.
I will check and reply to you as soon as possible!

File: backend/src-common/src/main/java/org/eclipse/sw360/datahandler/db/ComponentDatabaseHandler.java [hunk 1]
```diff
@@ -62,10 +62,21 @@
...
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
 import java.io.IOException;
 import java.io.InputStream;
+import java.io.InputStreamReader;
 import java.net.MalformedURLException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
```
Comment (akapti) [inline backend/src-common/src/main/java/org/eclipse/sw360/datahandler/db/ComponentDatabaseHandler.java:79]: Please remove all the unused imports.

**Review by akapti — CHANGES_REQUESTED** [review]
I have only reviewed the code.
will do the testing later.
Kindly incorporate the requested changes.

File: backend/src-common/pom.xml [hunk 2]
```diff
@@ -37,11 +37,24 @@
...
+        </dependency>
         <dependency>
             <groupId>org.spdx</groupId>
             <artifactId>spdx-tools</artifactId>
             <scope>compile</scope>
         </dependency>
+        <dependency>
+            <groupId>org.spdx</groupId>
+            <artifactId>tools-java</artifactId>
+            <version>1.0.2</version>
```
[the remark this answers is not in the document]
Comment (tienlee) [reply backend/src-common/pom.xml:55]: Thank you for your review.
We tried to upgrade the library but can't import SPDX files with 1.0.4. We are debugging to find the root cause.
Other comments were fixed.

[the remark this answers is not in the document]
Comment (akapti) [reply backend/src-common/pom.xml:55]: Thanks much @tienlee  .
I will also give it a try in my local environment with version `1.0.4`.

Comment (heliocastro) [inline backend/src-common/pom.xml:55]: Please add the versions on root pom.xml, and the dependencies on DependencyManagement.

[the remark this answers is not in the document]
Comment (akapti) [reply backend/src-common/pom.xml:55]: slack reference: https://sw360chat.slack.com/archives/C02NGJ2LX08/p1663221609110229

**Comment by tienlee** [comment]
@akapti and @heliocastro , we have committed to new changes. Please help me to check these changes.

**Comment by mcjaeger** [comment]
(from meeting:) needs re-review and re-test after PR comments have been implemented

**Review by akapti — CHANGES_REQUESTED** [review]
Suggested some minor review comments. Please incorporate.

Tested the feature and it's working fine without any issues.

**Comment by tienlee** [comment]
> Suggested some minor review comments. Please incorporate.
> 
> Tested the feature and it's working fine without any issues.

Thank you very much. We will fix these minor defects soon.

**Review by akapti — APPROVED** [review]
Code looks good to me.
Tested the changes and it's working as expected.

**Comment by akapti** [comment]
@tuannn2 @KoukiHama 
The existing ( and old SBOM import) functionality is such that whenever we import components from SBOM it does not set the `component type` field in component.
Since Component type is mandatory field in `UI` and `REST` API while creating the component.
It would be really good if we can set the `component type` to `OSS` while importing the components from SBOM. 

Kindly include this change in the current PR if possible, or else create a new PR.

**Comment by KoukiHama** [comment]
@akapti 

ok. I agree the idea about  components field

Could you merge this PR and make new issue and assign it to me?

**Comment by KoukiHama** [comment]
@akapti 

ok. I agree the idea about  components field

Could you merge this PR and make new issue and assign it to me?

we want to do it as new issue.

**Comment by KoukiHama** [comment]
@tienlee do you have idea about component type issue.

**Comment by tienlee** [comment]
> @tienlee do you have idea about component type issue.

I think this is a bug from the old version. Please create new issue and I will fix it.

**Comment by akapti** [comment]
> I think this is a bug from the old version. Please create new issue and I will fix it.

Created an issue #1658 . It's an already existing issue.
You can work on the above issue once this PR is merged.

**Pull request merged**